### PR TITLE
Feature/condo/sberdoma 366/add employee create/view

### PR DESCRIPTION
**Pull request by Dimitreee**
Screenshot 2021-05-26 at 07 35 15
Screenshot 2021-05-26 at 07 22 59
Screenshot 2021-05-26 at 07 23 02

### CODE REVIEW

File: apps/condo/domains/common/components/containers/FormList.tsx [hunk 1]
```diff
@@ -305,6 +305,8 @@ interface IFormWithAction {
     validateTrigger?: string | string[]
     resetOnComplete?: boolean
     layout?: 'vertical' | 'horizontal'
+    colon?: boolean
+    formValuesToMutationDataPreprocessor?: (values: IFormValuesType) => IFormValuesType
```
Comment (Dimitreee) [inline apps/condo/domains/common/components/containers/FormList.tsx:309]: duplicate https://github.com/open-condo-software/condo/pull/128

File: apps/condo/domains/organization/components/EmployeeRoleSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React, { useMemo } from 'react'
```
Comment (Dimitreee) [inline apps/condo/domains/organization/components/EmployeeRoleSelect.tsx:1]: Duplicate https://github.com/open-condo-software/condo/pull/128

File: apps/condo/domains/organization/components/ErrorsContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import styled from '@emotion/styled'
```
Comment (Dimitreee) [inline apps/condo/domains/organization/components/ErrorsContainer.tsx:1]: Will be encapsulated at https://doma.atlassian.net/browse/SBERDOMA-378

File: apps/condo/domains/organization/hooks/useTableColumns.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
+import { identity } from 'lodash/util'
```
Comment (Dimitreee) [inline apps/condo/domains/organization/hooks/useTableColumns.tsx:1]: Will be encapsulated at https://doma.atlassian.net/browse/SBERDOMA-378

File: apps/condo/domains/organization/schema/OrganizationEmployee.js [hunk 5]
```diff
@@ -79,6 +79,10 @@ const OrganizationEmployee = new GQLListSchema('OrganizationEmployee', {
             knexOptions: { isNotNullable: false }, // Relationship only!
             kmigratorOptions: { null: true, on_delete: 'models.SET_NULL' },
         },
+        position: {
```
Comment (Dimitreee) [inline apps/condo/domains/organization/schema/OrganizationEmployee.js:82]: duplicate #128

File: apps/condo/domains/organization/utils/helpers.ts [hunk 6]
```diff
@@ -0,0 +1,211 @@
+import { SortOrder } from 'antd/es/table/interface'
```
Comment (Dimitreee) [inline apps/condo/domains/organization/utils/helpers.ts:1]: Will be encapsulated at https://doma.atlassian.net/browse/SBERDOMA-378

File: apps/condo/migrations/20210526030905-0013_auto_20210526_0309.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+// auto generated by kmigrator
```
Comment (Dimitreee) [inline apps/condo/migrations/20210526030905-0013_auto_20210526_0309.js:1]: duplicate #128, will be removed

File: apps/condo/lang/ru.json [hunk 8]
```diff
@@ -284,6 +288,8 @@
   "SignOut": "Выйти",
   "Started": "Процесс запущен",
   "StartRecovery": "Восстановить",
+  "AddItemUsingForm": "Черз форму",
```
Comment (dkoviazin) [inline apps/condo/lang/ru.json:291]: spelling

**Review by dkoviazin — APPROVED** [review]
Ok

**Comment by AntonAL** [comment]
I get following error, when I try to register a new user and specifying already taken email:
> Unhandled Runtime Error
> Error: [error] Unable to register user
Page: `/employee/create`
Maybe It's not critical now…

File: apps/condo/domains/organization/components/ErrorsContainer.tsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+import { Col } from 'antd'
+import React from 'react'
+import { colors } from '@condo/domains/common/constants/style'
+
+export const ErrorsWrapper = styled.div`
+  display: inline-block;
+  padding: 9px 16px;
+  border-radius: 8px;
+  background-color: ${colors.beautifulBlue[5]};
+`
```
Comment (AntonAL) [inline apps/condo/domains/organization/components/ErrorsContainer.tsx:12]: It seems, like it also should be extracted into `common` domain.
Create a task for it, please.

**Review by AntonAL — APPROVED** [review]

File: apps/condo/domains/organization/components/CreateEmployeeForm.tsx [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+    // @ts-ignore
+    const { organization } = useOrganization()
+    const router = useRouter()
+
+    const FullNameLabel = intl.formatMessage({ id: 'pages.auth.register.field.Name' })
+    const FullNamePlaceholder = intl.formatMessage({ id: 'field.FullName' })
+    const PositionLabel = intl.formatMessage({ id: 'employee.Position' })
+    const PhoneLabel = intl.formatMessage({ id: 'Phone' })
+    const EmailLabel = intl.formatMessage({ id: 'Email' })
+    const RoleLabel = intl.formatMessage({ id: 'employee.Role' })
```
Comment (pahaz) [inline apps/condo/domains/organization/components/CreateEmployeeForm.tsx:43]:
```suggestion
    const intl = useIntl()
    const InviteEmployeeLabel = intl.formatMessage({ id: 'employee.InviteEmployee' })

    const FullNameLabel = intl.formatMessage({ id: 'pages.auth.register.field.Name' })
    const FullNamePlaceholder = intl.formatMessage({ id: 'field.FullName' })
    const PositionLabel = intl.formatMessage({ id: 'employee.Position' })
    const PhoneLabel = intl.formatMessage({ id: 'Phone' })
    const EmailLabel = intl.formatMessage({ id: 'Email' })
    const RoleLabel = intl.formatMessage({ id: 'employee.Role' })

    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    const { organization } = useOrganization()
    const router = useRouter()
```

File: apps/condo/domains/organization/components/CreateEmployeeForm.tsx [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+            colon={false}
+            formValuesToMutationDataPreprocessor={(values) => {
+                const role = get(values, 'role')
+
+                if (role) {
+                    values.role = { id: String(role) }
+                }
+
+                return values
+            }}
```
Comment (pahaz) [inline apps/condo/domains/organization/components/CreateEmployeeForm.tsx:93]: is it really needed?

File: apps/condo/domains/organization/components/EmployeeRoleSelect.tsx [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+        const convertedOption = OrganizationEmployeeRole.convertGQLItemToFormSelectState(status)
+
+        if (convertedOption) {
+            const { value, label } = convertedOption
+
+            return (<Select.Option key={value} value={value} title={label}>{label}</Select.Option>)
+        }
+    }), [employeeRoles, organizationId])
+
+    return !error && (
```
Comment (pahaz) [inline apps/condo/domains/organization/components/EmployeeRoleSelect.tsx:23]: error is an object. is it really working?

File: apps/condo/domains/organization/hooks/useTableColumns.tsx [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+            <Space size={8} direction={'vertical'} align={'center'}>
+                {props.children}
+                {
+                    props.showClearButton && (
+                        <Button
+                            size={'small'}
+                            onClick={() => props.clearFilters()}
+                            type={'inlineLink'}
+                        >
+                            {intl.formatMessage({ id: 'filters.Reset' })}
```
Comment (pahaz) [inline apps/condo/domains/organization/hooks/useTableColumns.tsx:34]: m

**Review by pahaz — APPROVED** [review]

**Pull request merged**